optin/reporting/utils/Validation.js

```diff
@@ -0,0 +1,664 @@
+	'Indian/Maldives',
+	'Indian/Mauritius',
+	'Indian/Mayotte',
+	'Indian/Reunion',
+	'Iran',
+	'Israel',
+	'Jamaica',
+	'Japan',
+	'Kwajalein',
+	'Libya',
+	'MET',
+	'MST',
+	'MST7MDT',
+	'Mexico/BajaNorte',
+	'Mexico/BajaSur',
+	'Mexico/General',
+	'NZ',
+	'NZ-CHAT',
+	'Navajo',
+	'PRC',
+	'PST8PDT',
+	'Pacific/Apia',
+	'Pacific/Auckland',
+	'Pacific/Bougainville',
+	'Pacific/Chatham',
+	'Pacific/Chuuk',
+	'Pacific/Easter',
+	'Pacific/Efate',
+	'Pacific/Enderbury',
+	'Pacific/Fakaofo',
+	'Pacific/Fiji',
+	'Pacific/Funafuti',
+	'Pacific/Galapagos',
+	'Pacific/Gambier',
+	'Pacific/Guadalcanal',
+	'Pacific/Guam',
+	'Pacific/Honolulu',
+	'Pacific/Johnston',
+	'Pacific/Kiritimati',
+	'Pacific/Kosrae',
+	'Pacific/Kwajalein',
+	'Pacific/Majuro',
+	'Pacific/Marquesas',
+	'Pacific/Midway',
+	'Pacific/Nauru',
+	'Pacific/Niue',
+	'Pacific/Norfolk',
+	'Pacific/Noumea',
+	'Pacific/Pago_Pago',
+	'Pacific/Palau',
+	'Pacific/Pitcairn',
+	'Pacific/Pohnpei',
+	'Pacific/Ponape',
+	'Pacific/Port_Moresby',
+	'Pacific/Rarotonga',
+	'Pacific/Saipan',
+	'Pacific/Samoa',
+	'Pacific/Tahiti',
+	'Pacific/Tarawa',
+	'Pacific/Tongatapu',
+	'Pacific/Truk',
+	'Pacific/Wake',
+	'Pacific/Wallis',
+	'Pacific/Yap',
+	'Poland',
+	'Portugal',
+	'ROC',
+	'ROK',
+	'Singapore',
+	'Turkey',
+	'UCT',
+	'US/Alaska',
+	'US/Aleutian',
+	'US/Arizona',
+	'US/Central',
+	'US/East-Indiana',
+	'US/Eastern',
+	'US/Hawaii',
+	'US/Indiana-Starke',
+	'US/Michigan',
+	'US/Mountain',
+	'US/Pacific',
+	'US/Pacific-New',
+	'US/Samoa',
+	'UTC',
+	'Universal',
+	'W-SU',
+	'WET',
+	'Zulu'
+];
+
+const Validation = {
+	validate: method => {
+		const paginationValidation = [
+			query('pageSize', 'pageSize required').exists().isInt(),
+			query('pageNumber', 'pageNumber required').exists().isInt()
+		];
+
+		const sortValidation = [
+			query('sortBy', 'field name must be string')
+				.optional().isString(),
+			query('sortDirection', 'sort direction is not supported')
+				.optional({ checkFalsy: true }).isIn(['asc', 'desc'])
+		];
+
+		const addressTransactionHashValidation = [
+			query('nemAddress', 'Nem address must be 40 character long')
+				.optional({ checkFalsy: true }).isString().isLength({ max: 40, min: 40 }),
+			query('transactionHash', 'Transaction hash must be 64 character long')
+				.optional({ checkFalsy: true }).isString().isLength({ max: 64, min: 64 })
+		];
+
+		switch (method) {
+		case 'getCompleted': {
+			return [
+				...paginationValidation,
+				...sortValidation,
+				...addressTransactionHashValidation,
+				query('optinType', 'optin type must be pre or post')
+					.optional({ checkFalsy: true }).isIn(['pre', 'post']),
+				query('symbolAddress', 'Symbol address must be 39 character long')
+					.optional({ checkFalsy: true }).isString().isLength({ max: 39, min: 39 })
+
+			];
+		}
+
+		case 'exportCsv': {
+			return [
+				query('timezone', 'timezone must be string example America/Los_Angeles').exists().isIn(validMomentTimezones)
+			];
+		}
+
+		case 'getOptinRequests': {
```

Comment:
I'd move page-specific validations next to `Controller`s rather than a utility file. So `getCompleted` and `getOptinRequests` can go next to (or inside) the controllers and the rest would stay here as shared validationUtility, _if that makes sense_

Reply:
The reason I put all validation in 1 place, is because it will be easier to manage IMO.

Some of the validation rules are shared between `getCompleted` and `getOptinRequests`.
if we move to page-specific validations `Controllers`, we may have some duplicated code.